### PR TITLE
<Proposal> Optimize the pods recovery efficiency when edge nodes restart proposal

### PR DESCRIPTION
**Pull request by Sodawyx**
…roposal

#### What this PR does / why we need it:
Optimize the pods recovery efficiency when edge nodes restart proposal
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: docs/proposals/20220725-pod-recovery-efficiency-proposal.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+- Promote the restart efficiency and make the cost time less than 30s
+- The solutions can be used in most hardwares
+
+## Question Analysis
+
+Kubelet is the controller of the Kubernetes work node. It is responsible for the creation and management of Pods on the work node. For an OpenYurt cluster, the components are the pods deployed on the Kubernetes cluster. When the work node restarts, Kubelet will recover the pods (Openyurt components) and the service pods. Therefore, detect the time delay of the Kubelet operations should be the focus.
+
+![](../img/pod-restart-step.png)
+
+The operations of Kubelet component when the work node restart are shown in Figure X. After the work node restart, Kubelet will initialize first. Then, Kubelet will start the static pods, for instance, YurtHub, etc. After that, YurtHub start-up and load the local cache. In the weak connection condition of the network, the Kubelet list/watch the pods from the YurtHub cache. Lastly, the Kubelet will recover the other pods according to the cache.
```
Comment (rambohe-ch) [inline docs/proposals/20220725-pod-recovery-efficiency-proposal.md:22]: which figure is the `Figure X`?

**Comment by rambohe-ch** [comment]
@Sodawyx Thank you for posting proposal. I think that it's not reasonable to optimize yurthub directly for this case. we need to measure the time for each step of node restart, then make decisions how to optimize based on the measure results.

**Comment by gnunu** [comment]
@Sodawyx Please consider comparing differebt restart scenrios to give optimization more lights:
1) node without yurthub/OpenYurt (must be connected in this case)
2) node connected

**Comment by rambohe-ch** [comment]
@Sodawyx please fix the markdown lint errors.

**Review by rambohe-ch — APPROVED** [review]

**Comment by rambohe-ch** [comment]
Fixes #858

**Pull request merged**